# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/compiler/translator/tree_ops/RewriteDfdy.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/compiler/translator/tree_ops/RewriteDfdy.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/angle/checkout/src/compiler/translator/tree_ops/RewriteDfdy.h	Fri Nov 08 17:06:00 2019 +0000
@@ -0,0 +1,30 @@
+//
+// Copyright 2019 The ANGLE Project Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+// This mutating tree traversal flips the output of dFdy() to account for framebuffer flipping.
+//
+// From: dFdy(p)
+// To:   (dFdy(p) * viewportYScale)
+//
+// See http://anglebug.com/3487
+
+#ifndef COMPILER_TRANSLATOR_TREEOPS_FLIP_DFDY_H_
+#define COMPILER_TRANSLATOR_TREEOPS_FLIP_DFDY_H_
+
+class TIntermNode;
+class TIntermBinary;
+class TSymbolTable;
+
+namespace sh
+{
+
+void RewriteDfdy(TIntermNode *root,
+                 const TSymbolTable &symbolTable,
+                 int shaderVersion,
+                 TIntermBinary *viewportYScale);
+
+}  // namespace sh
+
+#endif  // COMPILER_TRANSLATOR_TREEOPS_FLIP_DFDY_H_
\ No newline at end of file